### PR TITLE
Adapt inputs and transfers to new MOOSE multiapp transfer syntax

### PR DESCRIPTION
**Pull request by GiudGiud**
See https://github.com/idaholab/moose/pull/19676

### CODE REVIEW

File: test/tests/cht/nondimensional/nek_master.i [hunk 1]
```diff
@@ -76,31 +76,27 @@
...
-    direction = to_multiapp
-    multi_app = nek
+    to_multi_app = nek
     variable = avg_flux
     source_boundary = '1'
   []
   [flux_integral]
     type = MultiAppPostprocessorTransfer
     to_postprocessor = flux_integral
-    direction = to_multiapp
```
Comment (aprilnovak) [inline test/tests/cht/nondimensional/nek_master.i:94]: Just curious, how does this work now? Here for example the postprocessor has the same name in parent + sub, how is the direction deduced?

Comment (aprilnovak) [reply test/tests/cht/nondimensional/nek_master.i:94]: Oh I didn't look close enough, I see `to/from_multi_app` now. Like the syntax!

**Comment by GiudGiud** [comment]
there's real failures here, I ll address them soon

**Comment by aprilnovak** [comment]
Thanks for this update! A few of the errors looked like random Nek failures (complicated Stack + caching sometimes just aborts a Nek test), but the OpenMC fails are definitely real. 

We'll also need to update the `tutorials` directory, which has a few tests.

**Comment by GiudGiud** [comment]
Did I break the openmc build? I m getting hdf5 compilation errors. 
I rebased let's see

I updated the tutorials

**Comment by aprilnovak** [comment]
Not sure, nothing looks obvious to me. I just re-triggered the test suite in a separate PR (which previously passed) to see if something changed on CIVET.

**Comment by GiudGiud** [comment]
It passed on the other recipe.
Seems the libmesh + petsc update in between broke the openmc build ?

**Comment by aprilnovak** [comment]
Hmm, okay - well it looks like the errors could be related to HDF5. I think the HDF5 changed within MOOSE recently, right?

**Comment by GiudGiud** [comment]
we now build HDF5 with petsc
This could be it. Seems we need to change the Cardinal recipe in some way then.
The Nek build wasnt using our petsc module right? Should we just have both builds not use it?

**Comment by aprilnovak** [comment]
Correct, I believe that both builds are using the script-based approach with PETSc.

**Comment by GiudGiud** [comment]
Yes but seems there's an accomodation `use_module = False` in the Nek recipe. 
Who created that recipe? @loganharbour ?

**Comment by aprilnovak** [comment]
Yes, @loganharbour created these recipes

**Comment by GiudGiud** [comment]
Ok this is on our radar. I d like to get the Cardinal VTB recipe running first.
We are still working on hdf5 things at the moment for MOOSE as well in parallel, so we should be in a good place to look at this next week

**Comment by loganharbour** [comment]
I believe that all we need now here is to update MOOSE again now that https://github.com/idaholab/moose/pull/20793 has made its way in. We may need to regold whatever is failing in order to produce it via exodus w/o HDF5.

**Comment by aprilnovak** [comment]
Looks like this is not giving the `string_view` error that we were encountering before:

```
/data/civet1/cardinal/contrib/moose/libmesh/installed/include/libmesh/mesh_base.h:811:44: error: 'std::string_view' has not been declared
  811 |   unsigned int get_elem_integer_index(std::string_view name) const;
```

**Comment by aprilnovak** [comment]
@ahuxford, would you please update the `NekRSStandaloneProblem` documentation once this is merged to be sure to reflect the new syntax changes to the MultiApp transfers? i.e. update the `multi_app` and `direction` parameters in the various code snippets

**Comment by aprilnovak** [comment]
I ended up just skipping the failing tests; they work for me locally, so there is just something about CIVET's environment that is having trouble with it. We need to get Cardinal updated ASAP so that others working on the Griffin-Nek coupling can move forward.

**Pull request merged**